backend/src/main/kotlin/org/jetbrains/kotlinconf/backend/Sessionize.kt

```diff
@@ -86,4 +89,35 @@ fun SessionizeData.toConference(): Conference {
     }
 
     return Conference(sessions, speakers)
+}
+
+internal fun List<Session>.mergeWorkshops(): List<Session> {
+    val (workshopParts, nonWorkshop) = partition { it.tags?.contains("Workshop") == true }
+
+    // Group by room
+    val workshopPartsByLocations = workshopParts.groupBy { it.location }
+    val workshops = workshopPartsByLocations.values.mapNotNull { parts ->
+        val startTime = parts.minOf { it.startsAt }
+        val endTime = parts.maxOf { it.endsAt }
+        val first = parts.find { it.title.contains("Part 1") } ?: return@mapNotNull null
+        val title = first.title.dropAfter(". Part").trim()
+
+        Session(
+            first.id,
+            title,
+            parts.joinToString("\n") { it.description },
+            parts.flatMap { it.speakerIds },
+            first.location,
+            startTime,
+            endTime,
+            first.tags
+        )
+    }
+
+    return nonWorkshop + workshops
+}
+
+internal fun String.dropAfter(chunk: String): String {
```

Comment:
I think this is https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.text/substring-before.html

Reply:
Thank you, will fix